enhancements/dns/externaldns-operator.md

```diff
@@ -0,0 +1,451 @@
+DNS requirements.
+
+To satisfy this, a cluster administrator could create 2 different ExternalDNS Operator Custom Resources, one for each cloud provider.
+
+#### Story 5
+
+As a cluster administrator, I want the ExternalDNS Operator to deploy one ExternalDNS instances for each of my DNS zones, with each deployment
+being configured to use a different set of credentials, since I do not want to reuse common credentials for both zones due to security and auditing concerns.
+
+### Implementation Details [optional]
+
+When using the ExternalDNS Route Source on OpenShift, ExternalDNS will create CNAME record for a
+Route's host field pointing to the Route status' `routerCanonicalHostname` field.
+An in-flight [PR for the Ingress Operator](https://github.com/openshift/cluster-ingress-operator/pull/610)
+will enhance the `routerCanonicalHostname` field so that the field has a unique domain that falls under
+the Ingress Controller's given domain (to ensure compatibility with existing wildcard DNS records for Ingress).
+
+As an alternative to using CNAME records, the ExternalDNS Route source implementation could point Route hostnames directly to
+LoadBalancer IP addresses or hostnames. This is not ideal since coordinating with administrator-managed load-balancers in this situation would be
+non-trivial. Using the `routerCanonicalHostname` field as a DNS middle-man (so to speak) simplifies ExternalDNS management.
+
+Note that the OpenShift [Route source implementation](https://github.com/kubernetes-sigs/external-dns/blob/master/source/ocproute.go)
+in ExternalDNS supports some of the common [ExternalDNS annotations](https://github.com/kubernetes-sigs/external-dns/blob/master/source/source.go),
+such as the ExternalDNS TTL annotation.
+
+Currently, the Ingress Operator publishes wildcard DNS Records to point an Ingress Controller's domain to the Ingress Controller's
+LoadBalancer type Service (or alternatively, a cluster administrator configures this for their administrator-managed Load Balancer, etc).
+In the future, the Ingress Operator could create DNSRecord Custom Resources to be managed by ExternalDNS (instead of the Ingress Operator's bespoke
+DNS provider). The Ingress Controller API could be expanded to include the ability for a cluster administrator to specify whether or not a Wildcard DNSRecord
+is desired.
+
+
+Here is an example of what an ExternalDNS deployment (created by the operator) would look like:
+
+```yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: external-dns
+spec:
+  strategy:
+    type: Recreate
+  selector:
+    matchLabels:
+      app: external-dns
+  template:
+    metadata:
+      labels:
+        app: external-dns
+    spec:
+      serviceAccountName: external-dns
+      containers:
+      - name: external-dns-private
+        image: k8s.gcr.io/external-dns/external-dns:v0.8.0
+        args:
+        - --source=openshift-route
+        - --provider=aws
```

Comment:
is this derived from what we see on the Infrastructure CR or are there situations where the provider may vary from where the cluster is hosted?

Reply:
This is mentioned further up in the `proposal` section.
Pasting here:
```
By default, the operator would assume the cluster's platform is the desired DNS Provider. The operator would allow a user to specify
an alternative External DNS Provider, such as BlueCat DNS. The ExternalDNS Operator would be able to create multiple ExternalDNS deployments
to run in concert, one for each ExternalDNS Operator Custom Resource (similar to how the OpenShift [Ingress Operator](https://github.com/openshift/cluster-ingress-operator)
```